### ISSUE TITLE
runner.py error: "No such file or directory: PosixPath('gradlew')"

### ISSUE THREAD

**Issue by danieldavidf** [JetBrains-Research/PyNose#4]
I followed the whole readme guide, but in the last step, when I ran the runner.py script, I got the following error:

![image](https://user-images.githubusercontent.com/14796469/133301235-7da8aa86-7a57-4e47-87b3-8c6697341efd.png)

**Comment by SmirnovOleg** [comment]
Hey, thanks for opening this issue!

Have you set the `PLUGIN_ROOT` parameter properly? It is on line 8, `runner.py`

**Comment by danieldavidf** [comment]
I believe so. I put the current directory, as specified _(the folder that contains README.md (this file), runner.py, and build.gradle)._

![image](https://user-images.githubusercontent.com/14796469/133347623-c95e8c79-0b04-4ae4-944d-694d7d00bba6.png)

Is that correct?

**Comment by WANGJIEKE** [comment]
Could you try using absolute path for those variables?

**Comment by danieldavidf** [comment]
Yeah, ´runner.py´ apparently worked well using absolute path. Now I have a problem running `get_csv_stats.py` script: 

![image](https://user-images.githubusercontent.com/14796469/133414270-8b3cd57b-3f64-40f1-9a22-a4a0a7954cf5.png)

**Comment by WANGJIEKE** [comment]
Could you post the content of `log.txt` here?

**Comment by danieldavidf** [comment]
[log.txt](https://github.com/JetBrains-Research/PyNose/files/7189801/log.txt)

**Comment by freddiewanah** [comment]
Hi I have the same problem here

**Comment by WANGJIEKE** [comment]
Is PyCharm running when you start the `runner.py` script? If so, please quit PyCharm completely before starting the `runner.py`.

The screenshot is a comparison between your `log.txt` (left side) and mine (right side). Looks like you already have PyCharm running and the plugin isn't starting (it didn't print args, paths, etc.).

![Snipaste_2021-09-21_11-05-32](https://user-images.githubusercontent.com/21030118/134224090-bbbd4292-caa0-4108-becb-b6a29c02723a.png)

**Comment by danieldavidf** [comment]
When I close PyCharm and run `runner.py`, it does not process beyond 0%:

![image](https://user-images.githubusercontent.com/14796469/134334449-68d81bca-9814-40f5-abbb-84b728025dd3.png)

In this run, PyCharm opened, but nothing happens. The log.txt file was generated with no content. I waited for more than 15 minutes and the percentage remained at 0%.

**Comment by WANGJIEKE** [comment]
Do you mean there is a PyCharm window showing up when running the `runner.py`? If so, could you please check if the lines 27-37 in the [`build.gradle`](https://github.com/JetBrains-Research/PyNose/blob/master/build.gradle#L27-L37) and line 17 in the [`plugin.xml`](https://github.com/JetBrains-Research/PyNose/blob/master/src/main/resources/META-INF/plugin.xml#L17) are uncommented? They should be uncommented after setting up the Python interpreter; otherwise the plugin won't start.

Make sure the [`build.gradle`](https://github.com/JetBrains-Research/PyNose/blob/master/build.gradle#L27-L37) and [`plugin.xml`](https://github.com/JetBrains-Research/PyNose/blob/master/src/main/resources/META-INF/plugin.xml#L17) look like this when running the `runner.py`.

![image](https://user-images.githubusercontent.com/21030118/134642189-967f8a68-c090-4bf7-abcb-4b13537bf2e3.png)

I just set up a Ubuntu 20.04 virtual machine and I didn't encounter any issue when running the tool

**Comment by danieldavidf** [comment]
Thanks for your attention. When I uncommented  lines 27-37 in the `build.gradle` and line 17 in the `plugin.xml`, the runner.py ran up to 100%.

![image](https://user-images.githubusercontent.com/14796469/134677127-dd74b863-410f-4119-8243-773e2c0cc4a9.png)

However, I could not find any generated JSON file. Here is the complete content of the newly generated `log.txt`:

project=PosixPath('/home/daniel/Documents/geokey/.git')
====STDOUT====
> Task :compileJava UP-TO-DATE
> Task :patchPluginXml
> Task :processResources
> Task :classes
> Task :instrumentCode
> Task :postInstrumentCode
> Task :jar
> Task :prepareSandbox

> Task :runIde
args = [pynose, /home/daniel/Documents/geokey/.git, /usr/bin/python3.8, /home/daniel/Documents/PyNose/output]
path = /home/daniel/Documents/geokey/.git
outputDir = /home/daniel/Documents/PyNose/output
outputFileName = /home/daniel/Documents/PyNose/output/.git.json
Cannot find specified Python interpreter; printing available ones...
pythonCandidate = Python 3.8: Python 3.8.10 (/usr/bin/python3.8)
If nothing printed, you may have to go to GUI mode to configure a Python interpreter

BUILD SUCCESSFUL in 10s
8 actionable tasks: 7 executed, 1 up-to-date

====STDERR====WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.ui.StartupUiUtil to constructor com.sun.java.swing.plaf.gtk.GTKLookAndFeel()
WARNING: Please consider reporting this to the maintainers of com.intellij.util.ui.StartupUiUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2021-09-24 09:46:18,484 [    563]   WARN - llij.ide.plugins.PluginManager - Resource bundle redefinition for plugin 'com.jetbrains.pycharm.community.customization'. Old value: messages.ActionsBundle, new value: messages.PyBundle 
2021-09-24 09:46:19,911 [   1990]   WARN - j.internal.DebugAttachDetector - Unable to start DebugAttachDetector, please add `--add-exports java.base/jdk.internal.vm=ALL-UNNAMED` to VM options 


project=PosixPath('/home/daniel/Documents/geokey/local_settings.example')
====STDOUT====
> Task :compileJava UP-TO-DATE
> Task :patchPluginXml UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :instrumentCode UP-TO-DATE
> Task :postInstrumentCode
> Task :jar UP-TO-DATE
> Task :prepareSandbox UP-TO-DATE

> Task :runIde
args = [pynose, /home/daniel/Documents/geokey/local_settings.example, /usr/bin/python3.8, /home/daniel/Documents/PyNose/output]
path = /home/daniel/Documents/geokey/local_settings.example
outputDir = /home/daniel/Documents/PyNose/output
outputFileName = /home/daniel/Documents/PyNose/output/local_settings.example.json
Cannot find specified Python interpreter; printing available ones...
pythonCandidate = Python 3.8: Python 3.8.10 (/usr/bin/python3.8)
If nothing printed, you may have to go to GUI mode to configure a Python interpreter

BUILD SUCCESSFUL in 8s
8 actionable tasks: 2 executed, 6 up-to-date

====STDERR====WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.ui.StartupUiUtil to constructor com.sun.java.swing.plaf.gtk.GTKLookAndFeel()
WARNING: Please consider reporting this to the maintainers of com.intellij.util.ui.StartupUiUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2021-09-24 09:46:27,184 [    612]   WARN - llij.ide.plugins.PluginManager - Resource bundle redefinition for plugin 'com.jetbrains.pycharm.community.customization'. Old value: messages.ActionsBundle, new value: messages.PyBundle 
2021-09-24 09:46:28,705 [   2133]   WARN - j.internal.DebugAttachDetector - Unable to start DebugAttachDetector, please add `--add-exports java.base/jdk.internal.vm=ALL-UNNAMED` to VM options 


project=PosixPath('/home/daniel/Documents/geokey/db')
====STDOUT====
> Task :compileJava UP-TO-DATE
> Task :patchPluginXml UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :instrumentCode UP-TO-DATE
> Task :postInstrumentCode
> Task :jar UP-TO-DATE
> Task :prepareSandbox UP-TO-DATE

> Task :runIde
args = [pynose, /home/daniel/Documents/geokey/db, /usr/bin/python3.8, /home/daniel/Documents/PyNose/output]
path = /home/daniel/Documents/geokey/db
outputDir = /home/daniel/Documents/PyNose/output
outputFileName = /home/daniel/Documents/PyNose/output/db.json
Cannot find specified Python interpreter; printing available ones...
pythonCandidate = Python 3.8: Python 3.8.10 (/usr/bin/python3.8)
If nothing printed, you may have to go to GUI mode to configure a Python interpreter

BUILD SUCCESSFUL in 7s
8 actionable tasks: 2 executed, 6 up-to-date

====STDERR====WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.ui.StartupUiUtil to constructor com.sun.java.swing.plaf.gtk.GTKLookAndFeel()
WARNING: Please consider reporting this to the maintainers of com.intellij.util.ui.StartupUiUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2021-09-24 09:46:35,630 [    574]   WARN - llij.ide.plugins.PluginManager - Resource bundle redefinition for plugin 'com.jetbrains.pycharm.community.customization'. Old value: messages.ActionsBundle, new value: messages.PyBundle 
2021-09-24 09:46:37,023 [   1967]   WARN - j.internal.DebugAttachDetector - Unable to start DebugAttachDetector, please add `--add-exports java.base/jdk.internal.vm=ALL-UNNAMED` to VM options 


project=PosixPath('/home/daniel/Documents/geokey/geokey')
====STDOUT====
> Task :compileJava UP-TO-DATE
> Task :patchPluginXml UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :instrumentCode UP-TO-DATE
> Task :postInstrumentCode
> Task :jar UP-TO-DATE
> Task :prepareSandbox UP-TO-DATE

> Task :runIde
args = [pynose, /home/daniel/Documents/geokey/geokey, /usr/bin/python3.8, /home/daniel/Documents/PyNose/output]
path = /home/daniel/Documents/geokey/geokey
outputDir = /home/daniel/Documents/PyNose/output
outputFileName = /home/daniel/Documents/PyNose/output/geokey.json
Cannot find specified Python interpreter; printing available ones...
pythonCandidate = Python 3.8: Python 3.8.10 (/usr/bin/python3.8)
If nothing printed, you may have to go to GUI mode to configure a Python interpreter

BUILD SUCCESSFUL in 13s
8 actionable tasks: 2 executed, 6 up-to-date

====STDERR====WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.ui.StartupUiUtil to constructor com.sun.java.swing.plaf.gtk.GTKLookAndFeel()
WARNING: Please consider reporting this to the maintainers of com.intellij.util.ui.StartupUiUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2021-09-24 09:46:43,497 [    600]   WARN - llij.ide.plugins.PluginManager - Resource bundle redefinition for plugin 'com.jetbrains.pycharm.community.customization'. Old value: messages.ActionsBundle, new value: messages.PyBundle 
2021-09-24 09:46:44,925 [   2028]   WARN - j.internal.DebugAttachDetector - Unable to start DebugAttachDetector, please add `--add-exports java.base/jdk.internal.vm=ALL-UNNAMED` to VM options

**Comment by WANGJIEKE** [comment]
Sorry for late reply. The log file says the tool cannot find the specified Python interpreter.

![image](https://user-images.githubusercontent.com/21030118/135778626-b1899b44-84f3-464c-80eb-71e678fd5749.png)

Please go to `runner.py` and see if the `PYTHON_INTERPRETER_NAME` is correctly configured.

**Comment by WANGJIEKE** [comment]
I also push a new branch. You may try to run the plugin in that branch. The new branch called "new-sdk-setup" and it improves how Python interpreter get set up and also will print more log for debug.

**Comment by SmirnovOleg** [comment]
By the way, any progress on this? @danieldavidf have you obtained any JSON?

**Comment by areyde** [comment]
@danieldavidf Yeah, sorry, we have also been pre-ocupied with some stuff! Please note that the version of tool from the paper is now located in the ASE2021 branch, we have significantly revamped the master, now it is more focused on work in the IDE. If you still need the help, I would be happy to get onto a call and help you!
